src/app/shared/shared.module.ts

```diff
@@ -1,9 +1,20 @@
 import { NgModule } from '@angular/core';
 
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+
 import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
+import { TimeDurationPipe } from './pipes/time-duration.pipe';
 
 @NgModule({
-  declarations: [ BreadcrumbsComponent ],
-  exports: [ BreadcrumbsComponent ],
+  imports: [ FontAwesomeModule ],
+  declarations: [
+    BreadcrumbsComponent,
+    TimeDurationPipe,
+  ],
+  exports: [
+    BreadcrumbsComponent,
+    TimeDurationPipe,
+    FontAwesomeModule,
```

Comment:
Is it the most efficient way to use `FontAwesomeModule`. Just wondering how it can affect final bundle size

Reply:
Fa-Icon component is not recognized by Angular when it is used in another component whose module doesn't import FontAwesomeModule. I don't think that it can hugely increase the size of bundle. As far as I know, it is considered to re-export modules in shared modules. There is even mentioning about it in Angular documentation. https://angular.io/guide/sharing-ngmodules